In [1]:
from google.colab import drive

In [3]:
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# Importing the libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as py
%matplotlib inline

In [8]:
#reading csv file from drive and creating a dataframe df
df=pd.read_csv('/content/drive/MyDrive/CSV Files/My Uber Drives - 2016.csv')

In [10]:
# Viewing the first 5 records of the dataframe
df.head(10)

START_DATE*        END_DATE* CATEGORY*           START*  \
0   1/1/2016 21:11   1/1/2016 21:17  Business      Fort Pierce   
1    1/2/2016 1:25    1/2/2016 1:37  Business      Fort Pierce   
2   1/2/2016 20:25   1/2/2016 20:38  Business      Fort Pierce   
3   1/5/2016 17:31   1/5/2016 17:45  Business      Fort Pierce   
4   1/6/2016 14:42   1/6/2016 15:49  Business      Fort Pierce   
5   1/6/2016 17:15   1/6/2016 17:19  Business  West Palm Beach   
6   1/6/2016 17:30   1/6/2016 17:35  Business  West Palm Beach   
7   1/7/2016 13:27   1/7/2016 13:33  Business             Cary   
8   1/10/2016 8:05   1/10/2016 8:25  Business             Cary   
9  1/10/2016 12:17  1/10/2016 12:44  Business          Jamaica   

             STOP*  MILES*         PURPOSE*  
0      Fort Pierce     5.1   Meal/Entertain  
1      Fort Pierce     5.0              NaN  
2      Fort Pierce     4.8  Errand/Supplies  
3      Fort Pierce     4.7          Meeting  
4  West Palm Beach    63.7   Customer Visit  
5  West Palm Beach     4.3   Meal/Entertain  
6       Palm Beach     7.1          Meeting  
7             Cary     0.8          Meeting  
8      Morrisville     8.3          Meeting  
9         New York    16.5   Customer Visit

In [12]:
# Last 5 rows
df.tail()

START_DATE*         END_DATE* CATEGORY*            START*  \
1151  12/31/2016 13:24  12/31/2016 13:42  Business           Kar?chi   
1152  12/31/2016 15:03  12/31/2016 15:38  Business  Unknown Location   
1153  12/31/2016 21:32  12/31/2016 21:50  Business        Katunayake   
1154  12/31/2016 22:08  12/31/2016 23:51  Business           Gampaha   
1155            Totals               NaN       NaN               NaN   

                 STOP*   MILES*        PURPOSE*  
1151  Unknown Location      3.9  Temporary Site  
1152  Unknown Location     16.2         Meeting  
1153           Gampaha      6.4  Temporary Site  
1154         Ilukwatta     48.2  Temporary Site  
1155               NaN  12204.7             NaN

In [17]:
# Understanding my dataframe
print(df.shape)
df.size

(1156, 7)


8092

In [19]:
# Viewing the statistics of my dataframe
print(df.describe())

             MILES*
count   1156.000000
mean      21.115398
std      359.299007
min        0.500000
25%        2.900000
50%        6.000000
75%       10.400000
max    12204.700000


In [21]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1156 entries, 0 to 1155
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   START_DATE*  1156 non-null   object 
 1   END_DATE*    1155 non-null   object 
 2   CATEGORY*    1155 non-null   object 
 3   START*       1155 non-null   object 
 4   STOP*        1155 non-null   object 
 5   MILES*       1156 non-null   float64
 6   PURPOSE*     653 non-null    object 
dtypes: float64(1), object(6)
memory usage: 63.3+ KB
None


1 numerical and 6 categorical fields in my dataset
and PURPOSE column has many NaN values (need to handle the missing values, we could either fill it with mean/mode after analysis or remove them)

Since PURPOSE is categorical and important for analysis, we will take the mode(maximum count of a val) and replace NaN, so let me find the most occurring one with value counts

In [23]:
df['PURPOSE*'].value_counts()

PURPOSE*
Meeting            187
Meal/Entertain     160
Errand/Supplies    128
Customer Visit     101
Temporary Site      50
Between Offices     18
Moving               4
Airport/Travel       3
Charity ($)          1
Commute              1
Name: count, dtype: int64

Since, Meeting is the most occurred, I will replace the missing values in the PURPOSE with Meeting, but if accuracy is not good, remove it.

For this analysis, lets remove it

In [28]:
#To check how many null values are there in the dataset in each column
df.isnull().sum()

START_DATE*      0
END_DATE*        1
CATEGORY*        1
START*           1
STOP*            1
MILES*           0
PURPOSE*       503
dtype: int64

In [32]:
# Removing null values
df=df.dropna()
df.isnull().sum()

START_DATE*    0
END_DATE*      0
CATEGORY*      0
START*         0
STOP*          0
MILES*         0
PURPOSE*       0
dtype: int64

In [34]:
df.shape

(653, 7)

The dataset now has 653 non null values

In [35]:
df.describe()

MILES*
count  653.000000
mean    11.196325
std     22.986429
min      0.500000
25%      3.200000
50%      6.400000
75%     10.400000
max    310.300000

In [44]:
print(df['START*'].unique())
print(len(df['START*'].unique()))

['Fort Pierce' 'West Palm Beach' 'Cary' 'Jamaica' 'New York' 'Elmhurst'
 'Midtown' 'East Harlem' 'Flatiron District' 'Midtown East'
 'Hudson Square' 'Lower Manhattan' "Hell's Kitchen" 'Downtown' 'Gulfton'
 'Houston' 'Eagan Park' 'Morrisville' 'Durham' 'Farmington Woods'
 'Lake Wellingborough' 'Fayetteville Street' 'Raleigh' 'Whitebridge'
 'Hazelwood' 'Fairmont' 'Meredith Townes' 'Apex' 'Chapel Hill'
 'Northwoods' 'Edgehill Farms' 'Eastgate' 'East Elmhurst'
 'Long Island City' 'Katunayaka' 'Colombo' 'Nugegoda' 'Unknown Location'
 'Islamabad' 'R?walpindi' 'Noorpur Shahan' 'Preston' 'Heritage Pines'
 'Tanglewood' 'Waverly Place' 'Wayne Ridge' 'Westpark Place' 'East Austin'
 'The Drag' 'South Congress' 'Georgian Acres' 'North Austin'
 'West University' 'Austin' 'Katy' 'Sharpstown' 'Sugar Land' 'Galveston'
 'Port Bolivar' 'Washington Avenue' 'Briar Meadow' 'Latta' 'Jacksonville'
 'Lake Reams' 'Orlando' 'Kissimmee' 'Daytona Beach' 'Ridgeland' 'Florence'
 'Meredith' 'Holly Springs' 'Chessingt

There are 131 unique starting points

In [46]:
print(df['STOP*'].unique())
print(len(df['STOP*'].unique()))

['Fort Pierce' 'West Palm Beach' 'Palm Beach' 'Cary' 'Morrisville'
 'New York' 'Queens' 'East Harlem' 'NoMad' 'Midtown' 'Midtown East'
 'Hudson Square' 'Lower Manhattan' "Hell's Kitchen" 'Queens County'
 'Gulfton' 'Downtown' 'Houston' 'Jamestown Court' 'Durham' 'Whitebridge'
 'Raleigh' 'Umstead' 'Hazelwood' 'Westpark Place' 'Meredith Townes'
 'Leesville Hollow' 'Apex' 'Chapel Hill' 'Williamsburg Manor'
 'Macgregor Downs' 'Edgehill Farms' 'Walnut Terrace' 'Midtown West'
 'Long Island City' 'Jamaica' 'Unknown Location' 'Colombo' 'Nugegoda'
 'Katunayaka' 'Islamabad' 'R?walpindi' 'Noorpur Shahan' 'Heritage Pines'
 'Tanglewood' 'Waverly Place' 'Wayne Ridge' 'Northwoods'
 'Depot Historic District' 'West University' 'Congress Ave District'
 'Convention Center District' 'North Austin' 'The Drag' 'Coxville'
 'South Congress' 'Katy' 'Alief' 'Sharpstown' 'Sugar Land' 'Galveston'
 'Port Bolivar' 'Washington Avenue' 'Greater Greenspoint' 'Latta'
 'Jacksonville' 'Kissimmee' 'Lake Reams' 'Orlando' 'D

There are 137 unique stopping points

In [47]:
# Lets identify most popular starting points
df['START*'].value_counts()

START*
Cary                161
Unknown Location     55
Morrisville          54
Whitebridge          36
Durham               30
                   ... 
Lower Manhattan       1
Lake Reams            1
Latta                 1
Briar Meadow          1
Gampaha               1
Name: count, Length: 131, dtype: int64

So, based on the value counts on START, it is evident that Cary is the most popular starting point.

In [48]:
# Lets see the most popular destination
df['STOP*'].value_counts()

STOP*
Cary                155
Morrisville          60
Unknown Location     56
Whitebridge          37
Durham               30
                   ... 
Cedar Hill            1
Florence              1
Ridgeland             1
Daytona Beach         1
Ilukwatta             1
Name: count, Length: 137, dtype: int64

Cary is again the most popular destination

In [57]:
df2=df[df['START*']!='Unknown Location']
df2=df[df['STOP*']!='Unknown Location']

In [58]:
(df2['STOP*']=='Unknown Location').value_counts()

STOP*
False    597
Name: count, dtype: int64

So there are 26 unknown location records

In [77]:
# Grouping the start and stop to see which trip covered most miles
df2.groupby(['START*','STOP*'])['MILES*'].sum().sort_values(ascending=False).head(10)

START*        STOP*       
Cary          Durham          312.3
Latta         Jacksonville    310.3
Durham        Cary            298.4
Cary          Morrisville     293.7
Raleigh       Cary            269.5
Morrisville   Cary            250.6
Cary          Cary            233.9
              Raleigh         230.4
Jacksonville  Kissimmee       201.0
Boone         Cary            180.2
Name: MILES*, dtype: float64

Based on the grouping above, the person has travelled the most from Cary to Durham, covering 312 Miles

In [83]:
# Grouping start and stop pairs and checking size of the resulted dataframe to see which trip pair was most travelled
df2.groupby(['START*','STOP*']).size().sort_values(ascending=False)

START*         STOP*      
Cary           Morrisville    52
Morrisville    Cary           51
Cary           Cary           44
               Durham         30
Durham         Cary           29
                              ..
Jacksonville   Ridgeland       1
Katunayaka     Katunayaka      1
Katunayake     Gampaha         1
Katy           Houston         1
Winston Salem  Asheville       1
Length: 229, dtype: int64

Based on the above, most trips were between Cary and MorrisVille

In [92]:
import pandas as pd
from datetime import datetime

In [110]:
#Convering the columns from string format to date

df2['START_DATE*']=pd.to_datetime(df2['START_DATE*'])
df2['STOP_DATE*']=pd.to_datetime(df2['END_DATE*'])

<ipython-input-110-3d29c2c87210>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['START_DATE*']=pd.to_datetime(df2['START_DATE*'])
<ipython-input-110-3d29c2c87210>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['STOP_DATE*']=pd.to_datetime(df2['END_DATE*'])


In [98]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 597 entries, 0 to 1154
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   START_DATE*  597 non-null    object        
 1   END_DATE*    597 non-null    object        
 2   CATEGORY*    597 non-null    object        
 3   START*       597 non-null    datetime64[ns]
 4   STOP*        597 non-null    datetime64[ns]
 5   MILES*       597 non-null    float64       
 6   PURPOSE*     597 non-null    object        
dtypes: datetime64[ns](2), float64(1), object(4)
memory usage: 53.5+ KB


In [111]:
#Creating a new column to calculate duration taken for each trip

df2['DURATION*']=df2['STOP_DATE*']-df2['START_DATE*']
df2.head()

<ipython-input-111-0c4a455bf260>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['DURATION*']=df2['STOP_DATE*']-df2['START_DATE*']


START_DATE*       END_DATE* CATEGORY*              START*  \
0 2016-01-01 21:11:00  1/1/2016 21:17  Business 2016-01-01 21:11:00   
2 2016-01-02 20:25:00  1/2/2016 20:38  Business 2016-01-02 20:25:00   
3 2016-01-05 17:31:00  1/5/2016 17:45  Business 2016-01-05 17:31:00   
4 2016-01-06 14:42:00  1/6/2016 15:49  Business 2016-01-06 14:42:00   
5 2016-01-06 17:15:00  1/6/2016 17:19  Business 2016-01-06 17:15:00   

                STOP*  MILES*         PURPOSE*       DURATION*  \
0 2016-01-01 21:17:00     5.1   Meal/Entertain 0 days 00:06:00   
2 2016-01-02 20:38:00     4.8  Errand/Supplies 0 days 00:13:00   
3 2016-01-05 17:45:00     4.7          Meeting 0 days 00:14:00   
4 2016-01-06 15:49:00    63.7   Customer Visit 0 days 01:07:00   
5 2016-01-06 17:19:00     4.3   Meal/Entertain 0 days 00:04:00   

           STOP_DATE*  
0 2016-01-01 21:17:00  
2 2016-01-02 20:38:00  
3 2016-01-05 17:45:00  
4 2016-01-06 15:49:00  
5 2016-01-06 17:19:00

In [112]:
#Change the duration to minutes

df2['DURATION*']=df2['DURATION*'].apply(lambda x: pd.Timedelta.to_pytimedelta(x).days/(24*60)+pd.Timedelta.to_pytimedelta(x).seconds/60)

<ipython-input-112-120da23d3a31>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['DURATION*']=df2['DURATION*'].apply(lambda x: pd.Timedelta.to_pytimedelta(x).days/(24*60)+pd.Timedelta.to_pytimedelta(x).seconds/60)


In [114]:
df2['DURATION*'].head()

0     6.0
2    13.0
3    14.0
4    67.0
5     4.0
Name: DURATION*, dtype: float64

In [109]:
df2.groupby(['START*','STOP*'])['DURATION*'].sum()

START*               STOP*              
2016-01-01 21:11:00  2016-01-01 21:17:00      6.0
2016-01-02 20:25:00  2016-01-02 20:38:00     13.0
2016-01-05 17:31:00  2016-01-05 17:45:00     14.0
2016-01-06 14:42:00  2016-01-06 15:49:00     67.0
2016-01-06 17:15:00  2016-01-06 17:19:00      4.0
                                            ...  
2016-12-30 16:45:00  2016-12-30 17:08:00     23.0
2016-12-30 23:06:00  2016-12-30 23:10:00      4.0
2016-12-31 01:07:00  2016-12-31 01:14:00      7.0
2016-12-31 21:32:00  2016-12-31 21:50:00     18.0
2016-12-31 22:08:00  2016-12-31 23:51:00    103.0
Name: DURATION*, Length: 596, dtype: float64

In [117]:
df2['DURATION*'].describe()

count    597.000000
mean      22.922948
std       25.788324
min        2.000000
25%       11.000000
50%       17.000000
75%       26.000000
max      330.000000
Name: DURATION*, dtype: float64

In [118]:
# By mistake we have changed the START* to start date.
df2.head()

START_DATE*       END_DATE* CATEGORY*              START*  \
0 2016-01-01 21:11:00  1/1/2016 21:17  Business 2016-01-01 21:11:00   
2 2016-01-02 20:25:00  1/2/2016 20:38  Business 2016-01-02 20:25:00   
3 2016-01-05 17:31:00  1/5/2016 17:45  Business 2016-01-05 17:31:00   
4 2016-01-06 14:42:00  1/6/2016 15:49  Business 2016-01-06 14:42:00   
5 2016-01-06 17:15:00  1/6/2016 17:19  Business 2016-01-06 17:15:00   

                STOP*  MILES*         PURPOSE*  DURATION*          STOP_DATE*  
0 2016-01-01 21:17:00     5.1   Meal/Entertain        6.0 2016-01-01 21:17:00  
2 2016-01-02 20:38:00     4.8  Errand/Supplies       13.0 2016-01-02 20:38:00  
3 2016-01-05 17:45:00     4.7          Meeting       14.0 2016-01-05 17:45:00  
4 2016-01-06 15:49:00    63.7   Customer Visit       67.0 2016-01-06 15:49:00  
5 2016-01-06 17:19:00     4.3   Meal/Entertain        4.0 2016-01-06 17:19:00

In [119]:
#Reverting the start and stop columns to original values based on first DF
df2['START*']=df['START*']
df2['STOP*']=df['STOP*']

<ipython-input-119-985469a555b3>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['START*']=df['START*']
<ipython-input-119-985469a555b3>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['STOP*']=df['STOP*']


In [120]:
df2.head()

START_DATE*       END_DATE* CATEGORY*           START*  \
0 2016-01-01 21:11:00  1/1/2016 21:17  Business      Fort Pierce   
2 2016-01-02 20:25:00  1/2/2016 20:38  Business      Fort Pierce   
3 2016-01-05 17:31:00  1/5/2016 17:45  Business      Fort Pierce   
4 2016-01-06 14:42:00  1/6/2016 15:49  Business      Fort Pierce   
5 2016-01-06 17:15:00  1/6/2016 17:19  Business  West Palm Beach   

             STOP*  MILES*         PURPOSE*  DURATION*          STOP_DATE*  
0      Fort Pierce     5.1   Meal/Entertain        6.0 2016-01-01 21:17:00  
2      Fort Pierce     4.8  Errand/Supplies       13.0 2016-01-02 20:38:00  
3      Fort Pierce     4.7          Meeting       14.0 2016-01-05 17:45:00  
4  West Palm Beach    63.7   Customer Visit       67.0 2016-01-06 15:49:00  
5  West Palm Beach     4.3   Meal/Entertain        4.0 2016-01-06 17:19:00

In [127]:
# Adding a new column called Month to see which month was most preferred for trips

df2['MONTH*']=df2['START_DATE*'].dt.month_name()

<ipython-input-127-3acc6bde4067>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['MONTH*']=df2['START_DATE*'].dt.month_name()


In [130]:
df2.groupby(['START*','STOP*'])['MONTH*'].value_counts().sort_values(ascending=False)

START*         STOP*        MONTH*  
Kar?chi        Kar?chi      December    20
Cary           Cary         December    17
Lahore         Lahore       December    16
Cary           Cary         November    11
                            January     10
                                        ..
Katunayaka     Katunayaka   February     1
Katunayake     Gampaha      December     1
Katy           Houston      March        1
Kenner         New Orleans  June         1
Winston Salem  Asheville    October      1
Name: count, Length: 330, dtype: int64

In [131]:
df2['MONTH*'].value_counts()

MONTH*
December    101
June         72
March        71
February     68
January      59
November     59
April        50
May          46
July         46
October      19
August        6
Name: count, dtype: int64

In [144]:
df3=(df2['START*']=='Katy') & (df2['STOP*']=='Houston')
df3.value_counts().apply(lambda x: print('Yes, trip from Katy to Houston exists') if x==True else None)

Yes, trip from Katy to Houston exists


False    None
True     None
Name: count, dtype: object

In [149]:
df2['DAY NAME']=df2['START_DATE*'].dt.day_name()
df2['YEAR']=df2['START_DATE*'].dt.year
df2['DAY']=df2['START_DATE*'].dt.day
df2.head()

<ipython-input-149-653bd32103b2>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['DAY NAME']=df2['START_DATE*'].dt.day_name()
<ipython-input-149-653bd32103b2>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2['YEAR']=df2['START_DATE*'].dt.year
<ipython-input-149-653bd32103b2>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs

START_DATE*       END_DATE* CATEGORY*           START*  \
0 2016-01-01 21:11:00  1/1/2016 21:17  Business      Fort Pierce   
2 2016-01-02 20:25:00  1/2/2016 20:38  Business      Fort Pierce   
3 2016-01-05 17:31:00  1/5/2016 17:45  Business      Fort Pierce   
4 2016-01-06 14:42:00  1/6/2016 15:49  Business      Fort Pierce   
5 2016-01-06 17:15:00  1/6/2016 17:19  Business  West Palm Beach   

             STOP*  MILES*         PURPOSE*  DURATION*          STOP_DATE*  \
0      Fort Pierce     5.1   Meal/Entertain        6.0 2016-01-01 21:17:00   
2      Fort Pierce     4.8  Errand/Supplies       13.0 2016-01-02 20:38:00   
3      Fort Pierce     4.7          Meeting       14.0 2016-01-05 17:45:00   
4  West Palm Beach    63.7   Customer Visit       67.0 2016-01-06 15:49:00   
5  West Palm Beach     4.3   Meal/Entertain        4.0 2016-01-06 17:19:00   

    MONTH*  DAY  YEAR   DAY NAME  
0  January    1  2016     Friday  
2  January    2  2016   Saturday  
3  January    5  2016    Tuesday  
4  January    6  2016  Wednesday  
5  January    6  2016  Wednesday

In [152]:
# Checking which month and day were most travelled

df2['DAY NAME'].value_counts()

DAY NAME
Friday       119
Tuesday       87
Thursday      80
Monday        80
Sunday        79
Saturday      76
Wednesday     76
Name: count, dtype: int64

In [153]:
days={'Monday':0,'Tuesday':1,'Wednesday':2,'Thursday':3,'Friday':4,'Saturday':5,'Sunday':6}


In [160]:
# Checking avg miles travelled each month
df2.groupby(['MONTH*'])['MILES*'].mean().sort_values(ascending=False)

MONTH*
October     25.742105
April       21.898000
March       20.505634
November    10.655932
July        10.615217
January      8.486441
February     8.098529
May          7.793478
June         7.441667
December     5.315842
August       4.650000
Name: MILES*, dtype: float64

In [162]:
# How many hours in a month people have travelled

df2.groupby(['MONTH*'])['DURATION*'].sum()

MONTH*
April       1669.0
August        87.0
December    1840.0
February    1357.0
January     1090.0
July         936.0
June        1495.0
March       2262.0
May          696.0
November    1409.0
October      844.0
Name: DURATION*, dtype: float64

In [163]:
# Total miles travelled each month
df2.groupby(['MONTH*'])['MILES*'].sum().sort_values(ascending=False)

MONTH*
March       1455.9
April       1094.9
November     628.7
February     550.7
December     536.9
June         535.8
January      500.7
October      489.1
July         488.3
May          358.5
August        27.9
Name: MILES*, dtype: float64

**Now, lets see Categories**

In [164]:
df2['PURPOSE*'].value_counts()

PURPOSE*
Meeting            172
Meal/Entertain     152
Errand/Supplies    117
Customer Visit      95
Temporary Site      36
Between Offices     18
Moving               4
Charity ($)          1
Commute              1
Airport/Travel       1
Name: count, dtype: int64

In [166]:
df2.groupby(['PURPOSE*'])['DURATION*'].sum().sort_values(ascending=False)

PURPOSE*
Meeting            4879.0
Customer Visit     3222.0
Meal/Entertain     2413.0
Errand/Supplies    1496.0
Temporary Site      929.0
Between Offices     459.0
Commute             185.0
Moving               60.0
Charity ($)          27.0
Airport/Travel       15.0
Name: DURATION*, dtype: float64

Most hours travelled was for meetings

TypeError: datetime64 type does not support sum operations